cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/AppFabricServerTest.java

```diff
@@ -80,6 +103,24 @@ protected void configure() {
     AppFabricServer appFabricServer = injector.getInstance(AppFabricServer.class);
     appFabricServer.startAndWait();
     Assert.assertTrue(appFabricServer.isRunning());
+
+    Supplier<EndpointStrategy> endpointStrategySupplier = Suppliers.memoize(new Supplier<EndpointStrategy>() {
+      @Override
+      public EndpointStrategy get() {
+        return new RandomEndpointStrategy(discoveryServiceClient.discover(Constants.Service.APP_FABRIC_HTTP));
+      }
+    });
+    Discoverable discoverable = endpointStrategySupplier.get().pick(3, TimeUnit.SECONDS);
+    Assert.assertNotNull(discoverable);
+    Assert.assertArrayEquals(Constants.Security.SSL_DISCOVERABLE_KEY.getBytes(), discoverable.getPayload());
+    InetSocketAddress addr = discoverable.getSocketAddress();
+    // Since the server uses a self signed certificate we need a client that trusts all certificates
```

Comment:
what happens if we use a client that does not have a certificate? Is having requests from such clients fail a valid scenario? If so, should we verify that as well?